libs/telemetry/src/main/java/org/opensearch/telemetry/metrics/DefaultMetricsRegistry.java

```diff
@@ -34,6 +34,11 @@ public Counter createUpDownCounter(String name, String description, String unit)
         return metricsTelemetry.createUpDownCounter(name, description, unit);
     }
 
+    @Override
+    public Histogram createHistogram(String name, String description, String unit) {
```

Comment:
This is not the APIs we agreed upon:

```
Histogram createHistogram(String name, String description, String unit, List<Double> buckets);
```

Reply:
I think, we have discussed the other way round, to make it simple for now and let's not over complicated with additional paramaeters.

https://github.com/opensearch-project/OpenSearch/issues/10141#issuecomment-1912290093

Reply:
Also, like you mentioned in the RFC explicit bucketing is supported by only Otel not all other tools. If you think we should give more control to the users (which they might need because of their specific use cases and anomaly definitions) we can have both explicit bucket and dynamic support for users to choose from.

Only explicit buckets won't work and most of the time we don't know about the bucketing as it varies a lot based on the workloads and that's where the dynamic bucketing (exponential histogram in otel) scales well.

Reply:
> I think, we have discussed the other way round, to make it simple for now and let's not over complicated with additional paramaeters.

The best we can do here is probably to allow consumer to specify buckets, if the instrumentation does not support that - well, it cannot be integrated:

```
Histogram createHistogram(String name, String description, String unit, List<Double> buckets);
Histogram createHistogram(String name, String description, String unit);
```

Reply:
I agree with the strategy of allowing consumers to determine their preferred bucketing. However, clarity during development is often lacking, requiring tuning later on. When altering buckets, maintaining backward compatibility is crucial, especially in merging smaller ones into larger ones. To address this, Otel has introduced the exponential histogram, adjusting buckets based on values with a single scale factor for backward compatibility. However, not all sinks support exponential buckets, necessitating support for both dynamic (exponential) and fixed bucketing.

The clear, consumer-defined option is:
```
Histogram createHistogram(String name, String description, String unit, List<Double> buckets);
```

This default option can be configured for either dynamic (exponential histogram) or fixed bucketing with default buckets:
```
Histogram createHistogram(String name, String description, String unit);
```

Reply:
Most of the time the default behavior is going to be used. So we should build the strong default behavior and definitely the option to override the default behaviour.

Reply:
- >for histograms without buckets: document that the underlying framework will take care of it

- >for histograms with buckets: well, it is clear

I think, we both are on the same page. 

Just one more thing -

`for histograms without buckets: document that the underlying framework will take care of it`

Is it fine to provide a static config (framework level) to decide between dynamic (explicit histogram) vs fixed buckets (with default bucket) as all the sinks may not support different bucketing strategies.

Reply:
> Is it fine to provide a static config (framework level) to decide between dynamic (explicit histogram) vs fixed buckets (with default bucket) as all the sinks may not support different bucketing strategies.

No, it is not fine. I explained it here  https://github.com/opensearch-project/OpenSearch/pull/12062/files#r1473424829 . We should be very consistent in how things work during the **whole lifetime**, any changes like that will catch consumers by unpleasant surprise.  If we cannot be allowed.

Reply:
hmm. I think we are giving here the option while OpenSearch installation to decide the bucketing algorithm based on the observability tools users have and configure the default behavior. In the future, if otel or any other tool comes up with an new optimized bucketing strategy then how will we upgrade? Simply giving the option to choose this setting shouldn't be harmful and not cause the inconsistencies if decided thoughtfully.

Reply:
> hmm. I think we are giving here the option while OpenSearch installation to decide the bucketing algorithm based on the observability tools users have and configure the default behavior.

It is impossible to make this decision to satisfy everyone - there are many users, not one, of the system. Having consistent defaults for all is a good start. We could come to that later (if ever). For now, let's provide a simple API.

> In the future, if otel or any other tool comes up with an new optimized bucketing strategy then how will we upgrade? 

This is perfect example of what "depend on the instrumentation framework" means and this is what we should document.

Reply:
I have made the changes as discussed above.